src/components/media-controls/media-controls.spec.tsx

```diff
@@ -1,23 +1,27 @@
-/* eslint-disable no-undef */
 import { newSpecPage } from '@stencil/core/testing';
 import { jest } from '@jest/globals';
 // eslint-disable-next-line no-unused-vars
 import { h } from '@stencil/core';
-import { MediaException } from '../../utils/media';
 import { MediaControls } from './media-controls';
-import Logger from '../../utils/logger';
+import globals from '../../utils/globals';
+import { TestHTMLElement, TestHTMLMediaElement } from '../../test/types';
 
-const logger = new Logger();
-logger.warn = jest.fn();
-logger.log = jest.fn();
+beforeEach(() => {
+  let logger = globals._logger;
+  logger.log = jest.fn();
+  logger.warn = jest.fn();
+
+  //let win = globals._win;
+  globals._win.HTMLElement = TestHTMLElement;
+  globals._win.HTMLMediaElement = TestHTMLMediaElement;
```

Comment:
are you not actually modifying the original `window` reference here? Seems dangerous.

Also don't see how these are needed/used for this test.